### DISCUSSION TITLE
Allow customizing the `remoteEntry.mjs` file name

### DISCUSSION THREAD

**Discussion by gibsonjoshua55** [nrwl/nx#26784] [Feature Requests]
Most other aspects of builds are customizable through either the executor options or webpack config.  This file name, however is difficult to adjust.  Because it is set statically in the instantiation of the `ModuleFederationPlugin` in the `withModuleFederation` function, even wrapping that function doesn't easily allow that filename to be changed.

It would be great if we could set that filename in the `ModuleFederationConfig` options that we pass into that function.  That way if a particular project needs that file renamed, we can do so without an additional build steps or overriding configurations.

**Comment by timostroehlein** [comment]
We are currently solving this with the following workaround, it's not pretty but it works
```ts
import { withModuleFederation } from '@nx/module-federation/angular';
import moduleFederationConfig from './module-federation.config';

export default async (config: any) => {
  const fromModuleFederation = await withModuleFederation(
    moduleFederationConfig,
    { dts: false }
  );
  config = fromModuleFederation(config);
  // Workaround to change the name of the remoteEntry file
  config.plugins.find(
    (p: any) => p.constructor.name === 'ModuleFederationPlugin'
  )._options.filename = 'customRemoteEntry.js';
  return config;
};
```
